src/signatures.md

```diff
@@ -167,4 +167,59 @@ let card = BirthdayCard::new("Paul").age(64).text("Happy Valentine's Day!");
 # }
 ```
 
-Note another advantage of builders: Overloaded constructors often don't provide all possible combinations of parameters, whereas with the builder pattern, you can combine exactly the parameters you want.
\ No newline at end of file
+Note another advantage of builders: Overloaded constructors often don't provide all possible combinations of parameters, whereas with the builder pattern, you can combine exactly the parameters you want.
+
+## When must I use `#[must_use]`?
+
+> Use it on Results and mutex locks. - MG
+
+`#[must_use]` causes a compile error if the caller ignores the return value.
+
+Rust functions are often single-purpose. They either:
+
+* Return a value; or
+* Do something (i.e. have side effects) and return nothing.
+
+In neither case do you need to think about `#[must_use]`. (In the first case,
+nobody would call your function unless they were going to use the result.)
+
+`#[must_use]` is useful for those rarer functions which return a result _and_
+have side-effects. In most such cases, it's wise to specify `#[must_use]`, unless
```

Comment:
Omit hyphen in "side-effects" for consistency

Reply:
Done